### ISSUE TITLE
Bug: flat config errors miss important information

### ISSUE THREAD

**Issue by kkmuffme** [eslint/eslint#17370]
### Environment

Node version: 16.19.0
npm version: 8.9.0
Local ESLint version: 8.45.0
Global ESLint version: 8.45.0
Operating System: RHEL

### What parser are you using?

Default (Espree)

### What did you do?

Using the config exactly as provided here https://eslint.org/docs/latest/use/configure/configuration-files-new#using-configurations-included-in-plugins (but with require() instead of import, but that shouldn't matter)

Because jsdoc.configs.recommended.plugins is an array (that's a bug in jsdoc), it will fail. The output (with trace)
>TypeError: Key "plugins": Key "0": Expected an object.

Is useless however if you have a bigger eslint config or if the error is inherited, since there's no line number or anything to let me know where the issue is

### What did you expect to happen?

Error message with line number

### What actually happened?

>TypeError: Key "plugins": Key "0": Expected an object.

### Link to Minimal Reproducible Example

https://eslint.org/docs/latest/use/configure/configuration-files-new#using-configurations-included-in-plugins

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by fasttime** [comment]
Thanks! [The example](https://github.com/eslint/eslint/blob/1866e1df6175e4ba0ae4a0d88dc3c956bb310035/docs/src/use/configure/configuration-files-new.md?plain=1#L469-L486) is misleading because it's exporting what is currently a eslintrc-style config (`jsdoc.configs.recommended`) in a flat config array, which is not supported. This has been already noticed and is being discussed in #17355.

The thrown `TypeError` message seems unclear: if the rest of the team agrees, we could think about ways to improve it.

**Comment by nzakas** [comment]
I wouldn't say the message is useless, it lets you know that there's an invalid value in the `plugins` key. Finding the key is an issue because it is obscured as a reference.

Suggestions for how to improve the message are definitely welcome, I'm not just not sure what kind of message would clearly tell you that `jsdoc.configs.recommended` is the problem here.

**Comment by morganney** [comment]
It would be helpful to know if the schema violation is from a plugin or not, rather than user error, and therefore out of their hands.

**Comment by nzakas** [comment]
> Maybe the line number of where jsdoc.configs.recommended is used, so there is at least some starting point? Not sure if that's possible though

@kkmuffme we don't have that information. All ESLint gets is an array of objects.

> It would be helpful to know if the schema violation is from a plugin incompatibility with the new flat config or not, rather than user error, and therefore out of their hands.

@morganney I assume by "plugin compatibility" you mean if they're using the wrong config format? All plugins are, by default, compatible with flat config. It's the shareable configs that have compatibility issues.

In the case of the original comment, the error message is because eslintrc defines `plugins` as an array of strings whereas flat config defines `plugins` as an object.
---
After looking into this a bit, I think we can make a few changes to at least guess if someone is using eslintrc where they shouldn't be and explicitly mention that in the error message. We're never going to get to a point where we can point to the line in the config file that's causing the problem but maybe some more context would help.

**Comment by morganney** [comment]
> I assume by "plugin compatibility" you mean if they're using the wrong config format?

Yep.

I bumped into this issue with `eslint-plugin-jest` and ended up reconfiguring their plugin in my flat config to make things work.

```js
jest.configs.recommended.plugins = { jest }
jest.configs.recommended.languageOptions = {
  globals: {
    ...globals.jest
  }
}
delete jest.configs.recommended.env

export  default [
   jest.configs.recommended,
   {
      plugins: { jest }
   }
]
```
